modules/claims_api/app/controllers/claims_api/v2/veteran_identifier_controller.rb

```diff
@@ -5,18 +5,19 @@
 module ClaimsApi
   module V2
     class VeteranIdentifierController < ClaimsApi::V2::ApplicationController
-      # TODO: REMOVE BEFORE IMPLEMENTATION
-      skip_before_action :authenticate, only: %i[find]
-      ICN_FOR_TEST_USER = '1012667145V762142'
-      # TODO: REMOVE BEFORE IMPLEMENTATION
-
       def find
         raise ::Common::Exceptions::Unauthorized if request.headers['Authorization'].blank?
 
         validate_request
         veteran = find_veteran(params)
 
-        render json: { id: veteran[:id] }
+        raise ::Common::Exceptions::ResourceNotFound unless veteran_icn_found?(veteran)
+
+        user_is_the_veteran   = current_user_is_the_veteran?(veteran: veteran, user: @current_user)
+        user_is_a_veteran_rep = current_user_is_a_veteran_representative?(@current_user)
+        raise ::Common::Exceptions::Unauthorized unless user_is_the_veteran || user_is_a_veteran_rep
+
+        render json: { id: veteran.mpi.icn }
```

Comment:
I wanted to investigate jbuilder when we started implementing all this stuff. Worth looking into for this simple endpoint just as a concept for future larger complex endpoints?
https://github.com/rails/jbuilder

Reply:
Introduced and used "jbuilder" [here](https://github.com/department-of-veterans-affairs/vets-api/pull/7265/commits/729ce10f62d2a940f3f4e572e0e9a3f545543db8).